### DISCUSSION TITLE
Connecting to RDS from local machine (ssh tunnel?)

### DISCUSSION THREAD

**Discussion by coopsmoss** [aws/copilot-cli#2273] [Q&A] [answered]
I want to be able to manage my Aurora PostgreSQL instance from pgadmin or similar. 
Is there a recommended way to go about this? Google isn't very helpful. (or at least I'm googling the wrong things)
Usually you'd setup an ssh tunnel on the server but I'm not sure how that works in fargate container land. I'm not even sure how to ssh into it besides using `copilot svc exec`.
Do I need to deploy a docker image that hosts some kind of tunnel?

I know an EC2 instance could be setup to do this specifically, but I'm kind of wary of interfacing with copilot managed resources because I'm not confident about how they'll respond when/if changes are made. So I'd kind of like to keep this as copiloty as possible. 

Another option I've considered is deploying a pgadmin instance into the same environment as my rds instance, but that seems extreme.

Thanks for your help,
I'm really enjoying copilot thus far.

**Comment by efekarakus** [comment, marked as answer]
Hi @coopsmoss !  

One AWS native alternative might be to use the RDS Data API to be able to make queries against your database: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/data-api.html

You can enable this by modifying your addon template like this:
```yaml
myDBCluster:
    Type: 'AWS::RDS::DBCluster'
    Properties:
      EnableHttpEndpoint: true # Add this property and then run "copilot deploy"
```

Then in the console for RDS you can choose your database and "Query" it:
<img width="500" alt="Screen Shot 2021-05-06 at 8 38 45 AM" src="https://user-images.githubusercontent.com/879348/117326460-852a0780-ae46-11eb-8528-32dcbea08286.png">
and then in the query editor you'll be able to query against your database:
<img width="500" alt="Screen Shot 2021-05-06 at 8 43 51 AM" src="https://user-images.githubusercontent.com/879348/117327191-33ce4800-ae47-11eb-91ea-85c2dcdb2775.png">

Let me know if this route seems viable for you!

**Reply by coopsmoss** [reply]
This did work well thank you.

But my coworkers need a simpler solution involving a GUI. Someone pointed me towards a forwarding tool called port7777, though I did have to buy a licence, it should cover their use case.

Side note:
I wonder if you could build a simple tool that would forward a DB managers requests through the AWS cli using the RDS data api. I might play with that in the future.

Thank you for your help :)
